modules/imgproc/src/distransform.cpp

```diff
@@ -45,23 +45,21 @@ namespace cv
 {
 
 static const int DIST_SHIFT = 16;
-static const int INIT_DIST0 = INT_MAX;
-static const int DIST_MAX   = (INT_MAX >> 2);
 #define  CV_FLT_TO_FIX(x,n)  cvRound((x)*(1<<(n)))
 
 static void
-initTopBottom( Mat& temp, int border )
+initTopBottom( Mat& temp, int border, unsigned int value )
 {
     Size size = temp.size();
     for( int i = 0; i < border; i++ )
     {
-        int* ttop = temp.ptr<int>(i);
-        int* tbottom = temp.ptr<int>(size.height - i - 1);
+        unsigned int* ttop = temp.ptr<unsigned int>(i);
```

Comment:
> `.ptr<unsigned int>`

There is no support for 32U in `cv::Mat`.
This call is not valid.

---

`int` -> `unsigned int` may be not enough anyway.

Large images may require `uint64_t` to avoid overflows (very large images would still overflow anyway).
BTW, `uint64_t` is not supported by `Mat` too.

Reply:
As changes relate to temp May only, will it ok to replace on cv:;AutoBuffer or just an std::vector?

Reply:
Changed to `unsigned int*` cast from `int*`. At least the same performed at `unsigned int* tmp = (unsigned int*)(temp + BORDER*step) + BORDER;`